instill/model/v1alpha/model_service.proto

```diff
@@ -0,0 +1,202 @@
+    option (google.api.http) = {
+      patch : "/{model.name=models/*}"
+      body : "model"
+    };
+  }
+
+  // DeleteModel method receives a DeleteModelRequest message and returns a DeleteModelResponse
+  rpc DeleteModel(DeleteModelRequest) returns (DeleteModelResponse) {
+    option (google.api.http) = {
+      delete : "/{name=models/*}"
+    };
+  }
+
+  // ListModelInstance method receives a ListModelInstanceRequest message and returns a ListModelInstanceResponse
+  rpc ListModelInstance(ListModelInstanceRequest) returns (ListModelInstanceResponse) {
+    option (google.api.http) = {
+      get : "/{parent=models/*}/instances"
+    };
+  }
+
+  // CreateModelInstance method receives a CreateModelInstanceRequest message and returns a CreateModelInstanceResponse
+  rpc CreateModelInstance(CreateModelInstanceRequest) returns (CreateModelInstanceResponse) {
+    option (google.api.http) = {
+      post : "/{parent=models/*}/instances"
+      body : "instance"
+    };
+  }
+
+  // GetModelInstance method receives a GetModelInstanceRequest message and returns a GetModelInstanceResponse
+  rpc GetModelInstance(GetModelInstanceRequest) returns (GetModelInstanceResponse) {
+    option (google.api.http) = {
+    get : "/{name=models/*/instances/*}"
+    };
+  }
+
+  // UpdateModelInstance method receives a UpdateModelInstanceRequest message and returns a UpdateModelInstanceResponse
+  rpc UpdateModelInstance(UpdateModelInstanceRequest) returns (UpdateModelInstanceResponse) {
+    option (google.api.http) = {
+      patch : "/{instance.name=models/*/instances/*}"
+      body : "instance"
+    };
+  }
+
+  // DeleteModelInstance method receives a DeleteModelInstanceRequest message and returns a DeleteModelInstanceResponse
+  rpc DeleteModelInstance(DeleteModelInstanceRequest) returns (DeleteModelInstanceResponse) {
+    option (google.api.http) = {
+      delete : "/{name=models/*/instances/*}"
+    };
+  }
+
+  // GetModelInstanceCard method receives a GetModelInstanceCardRequest message and returns a GetModelInstanceCardResponse
+  rpc GetModelInstanceCard(GetModelInstanceCardRequest) returns (GetModelInstanceCardResponse) {
+    option (google.api.http) = {
+    get : "/{name=models/*/instances/*/readme}"
+    };
+  }
+
+  // UpdateModelInstanceCard method receives a UpdateModelInstanceCardRequest message and returns a UpdateModelInstanceCardResponse
+  rpc UpdateModelInstanceCard(UpdateModelInstanceCardRequest) returns (UpdateModelInstanceCardResponse) {
+    option (google.api.http) = {
+      patch : "/{readme.name=models/*/instances/*/readme}"
+      body: "readme"
+      };
+  }
+
+  ///////////////////////////////////////////////////////
+  //
+  // TriggerModel method receives a TriggerModelRequest message and
+  // returns a TriggerModelResponse message.
+  rpc TriggerModelInstance(TriggerModelInstanceRequest) returns (TriggerModelInstanceResponse) {
+    option (google.api.http) = {
+    post : "/{name=models/*/instances/*}:trigger"
```

Comment:
@xiaofei-du trigger is kind of action but our design follow entity. So should we use `:trigger` or `/outputs`

Reply:
After reading the Google API design [custom methods](https://google.aip.dev/136), we decide to adopt it and make `trigger a model instance` into a custom method.